experimenter/manifesttool/github_api.py

```diff
@@ -17,9 +17,33 @@
 
 def api_request(path: str, **kwargs: dict[str, Any]) -> dict[str, Any]:
     """Make a request to the GitHub API."""
-    return requests.get(
-        f"{GITHUB_API_URL}/{path}", headers=GITHUB_API_HEADERS, **kwargs
-    ).json()
+    url = f"{GITHUB_API_URL}/{path}"
+    rsp = requests.get(url, headers=GITHUB_API_HEADERS, **kwargs)
+
+    if rsp.status_code == 403:
+        if rsp.headers.get("X-RateLimit-Remaining") == "0":
+            raise Exception(f"Could not fetch {url}: GitHub API rate limit exceeded")
+
+    rsp.raise_for_status()
+
+    return rsp.json()
+
+
+def paginated_api_request(
+    path: str, per_page: int = 100
+) -> Generator[dict[str, Any], None, None]:
+    """Make several reqeusts to a paginated API resource and yield each page of results."""
+    page = 1
+
+    while True:
+        rsp = api_request(path, params={"page": page, "per_page": per_page})
+
+        if rsp:
```

Comment:
This is checking for the `{}` empty json response?

Reply:
Is there a way to make the conditions for the `break` case clearer?

Reply:
I added a comment.